packages/bot-skeleton/src/scratch/utils/index.js

```diff
@@ -7,6 +7,53 @@ import { removeLimitedBlocks } from '../../utils/workspace';
 import { saveWorkspaceToRecent } from '../../utils/local-storage';
 import DBotStore from '../dbot-store';
 import { log_types } from '../../constants/messages';
+import { error_message_map } from '../../utils/error-config';
+
+export const matchTranslateAttribute = translateString => {
+    const match = translateString.match(/translate\((-?\d*\.?\d+),(-?\d*\.?\d+)\)/);
+
+    if (match) {
+        const x = parseFloat(match[1]);
+        const y = parseFloat(match[2]);
+        return { x, y };
+    }
+    return null; // Invalid or no match
+};
+
+export const extractTranslateValues = () => {
+    const transform_value = Blockly?.derivWorkspace?.trashcan.svgGroup_.getAttribute('transform');
+    const translate_xy = matchTranslateAttribute(transform_value);
+
+    if (!translate_xy) {
+        globalObserver.emit('Error', 'Invalid String');
+    }
+
+    return {
+        translate_X: translate_xy.x,
+        translate_Y: translate_xy.y,
+    };
+};
+
+export const validateErrorOnBlockDelete = () => {
+    // Get the bounding rectangle of the selected block
+    const mandatory_blocks = ['trade_definition', 'before_purchase', 'during_purchase'];
```

Comment:
Can we put this in any of the config files where we are maintaining constant values? May be we can reuse this later?

Reply:
done ✅